dizoo/league_demo/league_scheduler.py

```diff
@@ -0,0 +1,142 @@
+        
+        assert schedule_mode in ['reduce','add','multi'], 'The schedule mode should be one of [\'reduce\', \'add\', \'multi\']'
+        self.schedule_mode = schedule_mode
+        
+        assert isinstance(change_amount, float) or isinstance(change_amount, int), 'The change_amount should be a float/int number greater than 0'
+        assert change_amount > 0, 'The change_amount should be a float/int number greater than 0'
+        self.change_amount = change_amount
+
+        assert isinstance(change_range, list) and len(change_range) == 2, 'The change_range should be a list with 2 float numbers'
+        assert (isinstance(change_range[0], float) or isinstance(change_range[0], int)) and (isinstance(change_range[1], float) or isinstance(change_range[1], int)), 'The change_range should be a list with 2 float/int numbers'
+        self.change_range = change_range
+
+        assert isinstance(threshold, float) or isinstance(threshold, int), 'The threshold should be a float/int number'
+        self.threshold = threshold
+
+        assert optimize_mode in ['min', 'max'], 'The optimize_mode should be one of [\'min\', \'max\']'
+        self.optimize_mode = optimize_mode
+
+        assert isinstance(patience, int), 'The patience should be a integer greater than or equal to 0'
+        assert patience >= 0, 'The patience should be a integer greater than or equal to 0'
+        self.patience = patience
+
+        assert isinstance(cooldown, int), 'The cooldown_counter should be a integer greater than or equal to 0'
+        assert cooldown >= 0, 'The cooldown_counter should be a integer greater than or equal to 0'
+        self.cooldown= cooldown
+        self.cooldown_counter = 0
+
+        self.last_metrics = None
+        self.bad_epochs_num = 0
+        
+
+    def step(self, metrics, cfg):
+        assert isinstance(metrics, float), 'The metrics should be a float number'
+        cur_metrics = metrics
+        
+        if self.is_better(cur_metrics):
+            self.bad_epochs_num = 0
+        else:
+            self.bad_epochs_num += 1
+        self.last_metrics = cur_metrics
+        
+        if self.in_cooldown:
+            self.cooldown_counter -= 1
+            self.bad_epochs_num = 0  # ignore any bad epochs in cooldown
+        
+        if self.bad_epochs_num > self.patience:
+            self.update_para(cfg)
+            self.cooldown_counter = self.cooldown
+            self.bad_epochs_num = 0
+    
+    def update_para(self, cfg):
+        if self.schedule_parameter == 'entropy_weight':
+            if self.schedule_mode == 'reduce':
+                tmp_parameter = cfg.policy.learn.entropy_weight - self.change_amount
+                if 
+                cfg.policy.learn.entropy_weight -= self.change_amount
+            if self.schedule_mode == 'add':
+                cfg.policy.learn.entropy_weight += self.change_amount
+            if self.schedule_mode == 'multi':
```

Comment:
Could we also have `div` here?

Reply:
My original thought is that if the factor is smaller than 1, then the 'multi' is exactly 'div', so i did not add 'div' mode here.

Reply:
If user want to perform div `3` on the `lr`, they don't want to write `0.33333333` in the multi mode, which is not good. Directly using div `3` is better.